zerver/migrations/0209_subscription_notifications_data_migration.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.18 on 2019-02-13 20:13
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+RECIPIENT_STREAM = 2
+DEFAULT_VALUES = {
+    'desktop_notifications': {
+        'default_for_streams': lambda user_profile: user_profile.enable_stream_desktop_notifications,
+        'default_for_pms': lambda user_profile: user_profile.enable_desktop_notifications,
+    },
+    'audible_notifications': {
+        'default_for_streams': lambda user_profile: user_profile.enable_stream_sounds,
+        'default_for_pms': lambda user_profile: user_profile.enable_sounds,
+    },
+    'push_notifications': {
+        'default_for_streams': lambda user_profile: user_profile.enable_stream_push_notifications,
+        'default_for_pms': lambda user_profile: (user_profile.enable_online_push_notifications or
+                                                 user_profile.enable_offline_push_notifications),
+    },
+    'email_notifications': {
+        'default_for_streams': lambda user_profile: user_profile.enable_stream_email_notifications,
+        'default_for_pms': lambda user_profile: user_profile.enable_offline_email_notifications,
+    },
+}
+
+def update_notification_settings(apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+    Subscription = apps.get_model('zerver', 'Subscription')
+
+    for sub in Subscription.objects.select_related('user_profile', 'recipient'):
```

Comment:
You can add `recipient__type_id__in=[1, 2]` here; it's a small perf optimization since we'll skip running this code on huddles.